src/pipx/commands/commands.py

```diff
@@ -48,30 +58,51 @@ def install(
     venv = Venv(venv_dir, python=python, verbose=verbose)
     try:
         venv.create_venv(venv_args, pip_args)
-        try:
-            venv.install_package(
-                package=package,
-                package_or_url=package_or_url,
-                pip_args=pip_args,
-                include_dependencies=include_dependencies,
-                include_apps=True,
-                is_main_package=True,
-            )
-        except PackageInstallFailureError:
-            venv.remove_venv()
-            raise PipxError(
-                f"Could not install package {package}. Is the name or spec correct?"
-            )
-
+        venv.install_package(
+            package=package_name,
+            package_or_url=package_spec,
+            pip_args=pip_args,
+            include_dependencies=include_dependencies,
+            include_apps=True,
+            is_main_package=True,
+        )
         _run_post_install_actions(
-            venv, package, local_bin_dir, venv_dir, include_dependencies, force=force
+            venv,
+            package_name,
+            local_bin_dir,
+            venv_dir,
+            include_dependencies,
+            force=force,
         )
     except (Exception, KeyboardInterrupt):
         print("")
         venv.remove_venv()
         raise
 
 
+def _package_name_from_spec(package_spec: str, python: str) -> str:
+    start_time = time.time()
+
+    package_name: Optional[str]
```

Comment:
I believe this is not Optional?

Reply:
Yes!  thanks for pointing it out, I had meant to delete that.